File: Backend/src/main/java/com/smartcity/backend/dto/User/AuthResponse.java
package com.smartcity.backend.dto.User;

import com.smartcity.backend.entitiy.Role;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AuthResponse {
    private String email;
    private String firstName;
    private String lastName;
    private Role role;
}

File: frontend-mobile/lib/data/models/auth_response.dart
class AuthResponse {
  final String email;
  final String firstName;
  final String lastName;
  final String role;

  AuthResponse({
    required this.email,
    required this.firstName,
    required this.lastName,
    required this.role,
  });

  factory AuthResponse.fromJson(Map<String, dynamic> json) => AuthResponse(
        email: json['email'],
        firstName: json['firstName'],
        lastName: json['lastName'],
        role: json['role'],
      );
}